utility/src/main/java/org/oppia/android/util/accessibility/BUILD.bazel

```diff
@@ -0,0 +1,32 @@
+load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_android_library")
+load("@rules_jvm_external//:defs.bzl", "artifact")
+
+kt_android_library(
+    name = "accessibility",
+    srcs = ["CustomAccessibilityManager.kt"],
+    visibility = ["//visibility:public"],
+)
+
+kt_android_library(
+    name = "impl",
```

Comment:
Unfortunately, I think I missed the circular dependency that arises here when we try to make ``accessiblity`` depend on ``impl``. There are a couple of different paths that we can take here, and I need to spend a bit more time thinking about this before proposing a specific solution. @fsharpasharp do you have a specific pattern that you think might work here?

Reply:
How about introducing a third library that exports the interface? None of the existing libraries will depend on it.

Reply:
That might unfortunately undermine the value of keeping impl separate (since it then exposes the public classes in impl which should stay private as part of impl). I think we might need to use a similar pattern where we export the modules of the package & have those depend on impl, but I'm not super keen on this pattern since it has its own drawbacks (such as requiring dependencies on all modules in central places like the android_binary or test targets).

Reply:
Okay, let's try the module + impl approach and see how that looks. @FareesHussain I suggest changing this to be:

```bazel
kt_android_library(
    name = "accessibility",
    visibility = ["//visibility:public"], 
    ... (source is interface only)
)

kt_android_library(
    name = "impl",
    ... (private visibility, source is accessibility module + prod implementation)
)

kt_android_library(
    name = "prod_module",
    visibility = ["//:oppia_binary_visibility"], 
    ... (source is new prod module -- see below; depends on :impl)
)

kt_android_library(
    name = "testing",
    visibility = ["//:oppia_testing_visibility"],
    testonly = True,
    ... (source is fake)
)

kt_android_library(
    name = "test_module",
    visibility = ["//:oppia_testing_visibility"],
    testonly = True,
    ... (source is new test module -- see below; depends on :testing)
)
```

You may also notice that there are both a test module & prod module. That's because I'm also suggesting that you make a functional change:
- Rename AccessibilityModule to be AccessibilityManagerModule (to clarify it's just for providing the manager)
- Introduce an AccessibilityProdModule that provides a binding of CustomAccessibilityManager to AndroidAccessibilityManager
- Introduce an AccessibilityTestModule that provides a binding of CustomAccessibilityManager to FakeAccessibilityManager
- Update all injections of AndroidAccessibilityModule to instead inject CustomAccessibilityManager so that it properly swaps out the accessibility version (you may be able to simplify some test bindings that are probably replacing the binding with the test version already)

Note also that the above requires introducing two new visibility groups: oppia_binary_visibility & oppia_testing_visibility to simplify visibility management. The former should work as-is, but the latter may not work until we split up the tests (in which case we should introduce the visibility such that it applies to all subpackages, and is later restricted to tests).

The nice thing about this package is that it actually includes all of the pieces we want to demonstrate in the pattern:
- A public API
- A production implementation that is kept private
- A production module that should use the private implementation & be incorporated in the top-level binary (or in tests that want to use the production module)
- A testing module that can optionally be used in tests
- A fake testing library

If you're unsure of all this, let me know. I'll introduce a new PR that replaces this one in that case since it may be easier for me to demonstrate the new pattern more explicitly that way.

Reply:
@fsharpasharp as well--what do you think of the above?

Reply:
@BenHenning I'll follow up with your PR regarding this pattern.